frontend/controller/actions/group.js

```diff
@@ -314,17 +314,29 @@ export default (sbp('sbp/selectors/register', {
   // secret keys to be shared with us, (b) ready to call the inviteAccept
   // action if we haven't done so yet (because we were previously waiting for
   // the keys), or (c) already a member and ready to interact with the group.
-  'gi.actions/group/join': async function (params: $Exact<ChelKeyRequestParams> & { options?: { skipUsableKeysCheck?: boolean; skipInviteAccept?: boolean } }) {
-    sbp('okTurtles.data/set', 'JOINING_GROUP-' + params.contractID, true)
+  'gi.actions/group/join': async function (params: $Exact<ChelKeyRequestParams> & { options?: { skipUsableKeysCheck?: boolean } }) {
     try {
       const rootState = sbp('state/vuex/state')
       const username = rootState.loggedIn.username
       const userID = rootState.loggedIn.identityContractID
 
       console.log('@@@@@@@@ AT join for ' + params.contractID)
-
+      const isJoiningGroup = !!params.originatingContractID
+      const preEffectBeforeSync = isJoiningGroup
+        ? () => { sbp('okTurtles.data/set', 'JOINING_GROUP-' + params.contractID, true) }
+        : () => {}
+      const postEffectAfterSync = isJoiningGroup
+        ? () => { sbp('okTurtles.data/set', 'JOINING_GROUP-' + params.contractID, false) }
+        : () => {}
+
+      preEffectBeforeSync()
```

Comment:
@Silver-IT It seems `preEffectBeforeSync` and `postEffectAfterSync` here are only used once in this block of code, I think it would feel more readable if these are just
```js
if (isJoiningGroup) {
  sbp('okTurtles.data/set', 'JOINING_GROUP-' + params.contractID, ...)
}
```
instead of making them as a function.

Reply:
Okay. I updated this way.
```
isJoiningGroup && sbp('okTurtles.data/set', 'JOINING_GROUP-' + params.contractID, true)
await sbp('chelonia/contract/sync', params.contractID)
```